3_javascript/javascript1/week3/script.js

```diff
@@ -0,0 +1,143 @@
+const freeCampCodeUser = "@sevda-89";
+console.log(freeCampCodeUser);
+
+const names = [
+  "Peter",
+  "Ahmad",
+  "Yana",
+  "kristina",
+  "Rasmus",
+  "Samuel",
+  "katrine",
+  "Tala",
+];
+const nameToRemove = "Ahmad";
+
+for (let i = 0; i < names.length; i++) {
+  if (names[i] === nameToRemove) {
+    names.splice(i, 1);
+    i--;
+  }
+}
+
+console.log(names);
+// ['Peter', 'Yana', 'kristina', 'Rasmus', 'Samuel', 'katrine', 'Tala']
+
+//When will we be there??
+
+function calculateTravelTime(travelInformation) {
+  const speed = travelInformation.speed;
+  const distance = travelInformation.destinationDistance;
+
+  // Calculate the time in hours and minutes
+  const hours = Math.floor(distance / speed);
+  const minutes = Math.floor((distance % speed) / (speed / 60));
+
+  // Format the time
+  const formattedTime = `${hours} hours and ${minutes} minutes`;
+
+  return formattedTime;
+}
+
+const travelInformation = {
+  speed: 50,
+  destinationDistance: 432,
+};
+
+const travelTime = calculateTravelTime(travelInformation);
+console.log(`I will arrive ${travelTime} later`);
+
+//Series duration of my life
+
+const seriesDurations = [
+  {
+    title: "Game of Thrones",
+    days: 3,
+    hours: 1,
+    minutes: 0,
+  },
+  {
+    title: "Sopranos",
+    days: 3,
+    hours: 14,
+    minutes: 0,
+  },
+  {
+    title: "The Wire",
+    days: 2,
+    hours: 12,
+    minutes: 0,
+  },
+];
+
+function percentageOfLifeSpentWatchedSeries(seriesDurations) {
+  const averageLifeInMinutes = 80 * 365 * 24 * 60;
+  // Converting 80 years to minutes
+
+  let totalMinutesSpent = 0;
+
+  for (const series of seriesDurations) {
+    const { days, hours, minutes } = series;
+    const seriesDurationInMinutes = days * 24 * 60 + hours * 60 + minutes;
+    totalMinutesSpent += seriesDurationInMinutes;
+  }
+
+  const percentageOfLifeSpent =
+    (totalMinutesSpent / averageLifeInMinutes) * 100;
+
+  return percentageOfLifeSpent;
+}
+
+function logOutSeriesText() {
+  for (const series of seriesDurations) {
+    const { title, days, hours, minutes } = series;
+    const seriesDurationInMinutes = days * 24 * 60 + hours * 60 + minutes;
+    const percentageOfLifeSpent =
+      (seriesDurationInMinutes / (80 * 365 * 24 * 60)) * 100;
+
+    console.log(`${title} took ${percentageOfLifeSpent}% of my life`);
+  }
+
+  const totalPercentage = percentageOfLifeSpentWatchedSeries(seriesDurations);
+  console.log(`It is total of ${totalPercentage}% of my life`);
+}
+
+logOutSeriesText();
+
```

Comment:
✅ The structure you used to store series information (seriesDurations array) is well-organized, making it easy to add or remove series in the future.
🎊 Calculating the percentage of life spent on each series individually provides good granularity.

❗ Areas of Improvement:

**Duplication**:
The calculation of life in minutes **(80 * 365 * 24 * 60)** is repeated. Using a **constant** like **AVERAGE_LIFE_IN_MINUTES** would be efficient.
**Precision**:
When displaying percentages, consider fixing the output to 2 decimal places for consistency and clarity.
**Refactoring**:
The calculation for the duration of a series in minutes is repeated. Consider a helper function:
`function calculateSeriesDurationInMinutes(series) {
    return series.days * 24 * 60 + series.hours * 60 + series.minutes;
}`